src/psyclone/profiler.py

```diff
@@ -169,12 +174,23 @@ def __init__(self, children=None, parent=None):
         # change every time gen() is called).
         self._var_name = NameSpaceFactory().create().create_name("profile")
 
-        # Name of the region. In general at constructor time we might not
-        # have a parent subroutine or a child for the kernel, so we leave
-        # the name empty for now. The region and module names are set the
-        # first time gen() is called (and then remain unchanged).
-        self._region_name = None
+        # Name of the region. In general at constructor time we might
+        # not have a parent subroutine or a child for the kernel, so
+        # the name is left empty, unless explicitly provided by the
+        # user. The region and module names are set the first time
+        # gen() is called (and then remain unchanged).
         self._module_name = None
+        self._region_name = None
+        if name:
+            if not isinstance(name, tuple) or not len(name) == 2 or \
+               not name[0] or not isinstance(name[0], str) or \
+               not name[1] or not isinstance(name[1], str):
+                raise GenerationError(
```

Comment:
I think this should be an InternalError as the creation of a ProfileNode is something that PSyclone is responsible for.

Reply:
Agreed.

Reply:
agreed.